File: evidence/generated-systems/angular/hr-on-angular/src/app/models/EmploymentContract.js

// Define collection and schema for EmploymentContract
export interface EmploymentContract {
    contractNumber:
	type : string
    startDate:
	type : Date
    endDate:
	type : Date
    workHoursPerWeek:
	type : String
    Employee:
	type : Schema.Types.ObjectId
    CompensationPackage:
	type : Schema.Types.ObjectId
    WorkSchedule:
	type : Schema.Types.ObjectId
    Location:
	type : Schema.Types.ObjectId
    PayrollCalendar:
	type : Schema.Types.ObjectId
    EmploymentType:
 	type : String
    Status:
 	type : String
    PayFrequency:
 	type : String
#
    collection: 'employmentContracts'
}

File: evidence/generated-systems/golang/hr-on-golang/hr-on-golang/internal/model/EmploymentContract.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// EmploymentContract Declaration
//==============================================================
type EmploymentContract struct {
    gorm.Model
     ContractNumber                                    string
    StartDate                                                            time.Time
    EndDate                                                            time.Time
    WorkHoursPerWeek                                                            string
    EmployeeId         *uint
    Employee           *Employee `gorm:"foreignKey:EmployeeId"`
    CompensationPackageId         *uint
    CompensationPackage           *CompensationPackage `gorm:"foreignKey:CompensationPackageId"`
    WorkScheduleId         *uint
    WorkSchedule           *WorkSchedule `gorm:"foreignKey:WorkScheduleId"`
    LocationId         *uint
    Location           *Location `gorm:"foreignKey:LocationId"`
    PayrollCalendarId         *uint
    PayrollCalendar           *PayrollCalendar `gorm:"foreignKey:PayrollCalendarId"`
    EmploymentType                      EmploymentType
    Status                      ContractStatus
    PayFrequency                      PayFrequency

// parent associations as their child

}

